src/data/fin/basic.lean

```diff
@@ -251,18 +251,25 @@ def factorial {n : ℕ} : fin n → ℕ
 instance {n : ℕ} : has_well_founded (fin n) :=
 ⟨_, measure_wf coe⟩
 
-@[simp] lemma coe_zero {n : ℕ} : ((0 : fin (n+1)) : ℕ) = 0 := rfl
+instance has_zero_of_ne_zero [ne_zero n] : has_zero (fin n) :=
+⟨⟨0, ne_zero.pos _⟩⟩
+
+instance has_one_of_ne_zero [ne_zero n] : has_one (fin n) :=
+⟨⟨1 % n, nat.mod_lt _ (ne_zero.pos _)⟩⟩
```

Comment:
You'll need to move it slightly lower in the file, but I think this makes sense:
```suggestion
instance has_one_of_ne_zero [ne_zero n] : has_one (fin n) :=
⟨of_nat' 1⟩
```